### DISCUSSION TITLE
How to compile a custom Linux kernel for use with QEMU

### DISCUSSION THREAD

**Discussion by ghost** [termux/termux-app#3563] [Q&A] [answered]
I am trying to compile a custom Linux kernel to be used by QEMU in Termux. But compilation fails. Here are the steps I follow:

pkg install build-essential ncurses bison flex openssl libelf clang
curl -LJO https://cdn.kernel.org/pub/linux/kernel/v6.x/linux-6.4.9.tar.xz
tar -xJvf linux-6.4.9.tar.xz
cd linux-6.4.9
make menuconfig

this fails with:

error: call to undeclared library function 'bcmp' with type 'int (const void *, const void *, unsigned long)'

it seems 'bcmp' is not supported in Termux as well as other functions. So my question is if it is possible to compile a custom Linux kernel under Termux.

**Comment by sylirre** [comment, marked as answer]
Compile it in `proot`. Linux kernel source has a lot of helper tools which were never tested on Android OS as compilation host.
